### PR TITLE
Merge type variable fixes and unit test

### PR DESCRIPTION
**Pull request by Arkatos1**
## About The Pull Request

This PR fixes a case where certain materials caused issues when working with stacking machines, because they did not have set merge_type from the get go, which meant that initial() of that variable returned null.

To clarify further - if /obj/item/stack **does not** have set merge_type, it is generated merge_type upon Initialize(), which is the same as its typepath. For example, currently /obj/item/stack/sheet/bluespace_crystal does not have any merge_type set, and it is given merge_type = /obj/item/stack/sheet/bluespace_crystal upon Initialize(). **Each** Initialize(). Again and again.

There are quite a bit of these cases in the codebase, especially if its some older code. I have gone through them and set all of them their set merge_type, which they would inevitably receive anyway upon initializing and it fixes a bug mentioned above.

To prevent this happening again, I have also included unit test to check if merge types are set for stacks, included exceptions are usually abstract paths like /obj/item/stack/sheet/mineral, which contains zero behavior on its own and does not spawn unless done via admin tools.

## Why It's Good For The Game

Bugfix and unit test.

## Changelog
:cl: Arkatos
fix: Fixed a case where certain materials could have problematic or missing behavior when interacting with stacking machines.
fix: Fixed a case where certain materials could not be merged together even if they were the same.
/:cl:

### CODE REVIEW

**Comment by Arkatos1** [comment]
Turns out there are actually 140 more bad cases to fix after running unit test.

**Review by Mothblocks — DISMISSED** [review]
Test is fine, DNM until it actually passes though.

File: code/modules/unit_tests/merge_type.dm [hunk 1]
```diff
@@ -1,5 +1,6 @@
 /datum/unit_test/merge_type/Run()
-	var/list/paths = subtypesof(/obj/item/stack) - /obj/item/stack/sheet - /obj/item/stack/sheet/mineral
+	var/list/paths = subtypesof(/obj/item/stack) - /obj/item/stack/sheet - /obj/item/stack/sheet/mineral - /obj/item/stack/ore
+	- /obj/item/stack/spacecash - /obj/item/stack/license_plates - /obj/item/stack/tile/mineral - /obj/item/stack/tile
```
Comment (Jared-Fogle) [inline code/modules/unit_tests/merge_type.dm:3]:
```suggestion
	var/list/blacklist = list(
		/obj/item/stack/sheet,
		/obj/item/stack/sheet/mineral,
		/obj/item/stack/ore,
		/obj/item/stack/spacecash,
		/obj/item/stack/license_plates,
		/obj/item/stack/tile/mineral,
		/obj/item/stack/tile,
	)
	var/list/paths = subtypesof(/obj/item/stack) - blacklist
```

Reads better and is easier to add onto.

Comment (Arkatos1) [reply code/modules/unit_tests/merge_type.dm]: Implemented just as you added review, heh.

**Comment by Arkatos1** [comment]
Conflicts resolved, all remaining bad cases fixed and PR is passsing all checks now. Unit test itself was also approved earlier. This should be ready to go.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**